Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -533,6 +542,26 @@ extension SQLiteBookmarks {
             (deleteStructure, args),
         ])
     }
+    
+    public func updateBookmarkByGUID(new: BookmarkItem) -> Success {
+        let args: Args = [
+            NSDate.nowNumber(),
+            new.url,
+            new.title,
+            SyncStatus.Changed.rawValue,
+            new.guid
+            ]
+        
+        let update =
+            "UPDATE \(TableBookmarksLocal) SET" +
+                " local_modified = ?" +
+                ", bmkUri = ?" +
+                ", title = ?" +
+                ", sync_status = ?" +
+                " WHERE guid = ?"
+        
+        return self.db.run([(update,args)])
```

Comment:
I'll let @rnewman correct me if I'm missing some details (which I most probably am) but I'll try to take a stab at this.

I think this needs to be more informed about making changes to the various underlying bookmark tables such as the folder structure and mirror tables. By using the way we update logins as a reference [1], a couple of things come to mind:

1. The bookmarks we display to the user are not just the local bookmarks but a result set from both the local and mirror tables (mirror containing all of our bookmarks that live on the server). For example, when fetching bookmarks from the `.Local` direction, we use the SQL view which combines the two.

2. Along with the split across local/mirror tables, we also have separate tables for storing the hierarchy of the bookmarks. I don't think changes to this are needed if we're modifying the _properties_ of a bookmark but when we start to modify the _position_ and folder a bookmark lives in we'll definitely need to make changes there.

3. Looking at the logins implementation, there seems to be some differentiation between modifying `.New` entries vs `.Changed` entries. Something else we might need to consider.

I think that when considering making changes to updating a bookmark, it would make more sense to address all possible modifications (moving, editing) vs just editing. I feel the logic we would write for updating an edit would eventually get replaced by the more complicated version anyways. It would be wise to split this chunk of work off into it's own thing separate from this PR since dealing with the intricacies of the bookmark sync engine is a beast of it's own and needs to be well tested.

[1] https://github.com/mozilla/firefox-ios/blob/master/Storage/SQL/SQLiteLogins.swift#L467
[2] https://github.com/mozilla/firefox-ios/blob/master/Storage/SQL/SQLiteBookmarksModel.swift#L296